spotify: handle MP3_160_ENC

theSmallNothing opened the pull request:
add lib/crypto.js: handles all key generation and decryption functionality
modify lib/spotify.js: handles sneaky `album_art` messages
modify lib/track.js: `play` now includes decryption transformer in pipeline
modify lib/util.js: added hex2bin and bin2hex helper functions
modify package.json: commit changes require newer `superagent`

Commit:
spotify: handle MP3_160_ENC

add lib/crypto.js: handles all key generation and decryption functionality
modify lib/spotify.js: handles sneaky `album_art` messages
modify lib/track.js: `play` now includes decryption transformer in pipeline
modify lib/util.js: added hex2bin and bin2hex helper functions
modify package.json: commit changes require newer `superagent`

Swyter commented on lib/spotify.js:
```diff
@@ -438,6 +452,13 @@ Spotify.prototype._onmessagecommand = function (command, args) {
+  } else if ('album_art' == command) {
+    var js = args[0][1];
+    try {
+      vm.runInContext("window = {eval: eval}; " + js, this._context);
```
Now, this is some album art I can get behind!

Commit:
spotify: remove unmaintained protobuf dependency

modify package.json: remove references to `protobuf`
modify lib/schemas.js: remove references to `protobuf`, make `protobufjs` the only parser
remove proto/*.desc: only required for the `protobuf` module

Commit:
Merge branch 'master' of git://github.com/mattyway/node-spotify-web

Lordmau5 committed:
Fix for trying to fetch a whole album / playlist at once.

So as it turns out, the old code seems to have taken the very first "album_art" message over to every single track that followed, if an Album or Playlist was fetched.
This code basically runs it through a new Context within the code-execution VM every time one of those messages is being received.

Lordmau5 committed:
Fix library method not being able to fetch saved tracks

Lordmau5 committed:
Fix indentation

Commit:
Merge pull request #1 from Lordmau5/master

Fix for trying to fetch a whole album / playlist at once.
Fix library method not being able to fetch saved tracks.
